libp2p/protocols/pubsub/pubsub.nim

```diff
@@ -183,11 +219,9 @@ method unsubscribePeer*(p: PubSub, peerInfo: PeerInfo) {.base, async.} =
     let peer = p.peers[peerInfo.id]
 
     trace "unsubscribing from peer", peerId = $peerInfo
-    if not(isNil(peer.conn)):
+    if not(isNil(peer)) and not(isNil(peer.conn)):
```

Comment:
here however, as opposed to https://github.com/status-im/nim-libp2p/pull/279/files/378544033e73472db4761c527c4ba7bf7a5a0e0f#r456219385, can happen that the peer or connection are simply nil on arrival, which whould indicate an issue upstream, or the peer simply disconnected in the process of trying to establish a pubsub connection.